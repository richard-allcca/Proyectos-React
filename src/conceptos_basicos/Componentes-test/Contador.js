import React, { useState } from "react";

const Contador = ({ valor = 100 }) => {
  const [value, setValue] = useState(valor);
  const sumar = () => setValue(value + 1);
  const restar = () => setValue(value - 1);
  const reset = () => setValue(valor);

  return (
    <>
      <h2>Contador</h2>
      <section className="contador">
        <button onClick={sumar}>+</button>
        <p>{value}</p>
        <button onClick={restar}>-</button>
      </section>
      <button onClick={reset}>Reset</button>
    </>
  );
};

export default Contador;
