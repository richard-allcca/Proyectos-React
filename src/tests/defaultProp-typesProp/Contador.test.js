import { shallow } from "enzyme";
import React from "react";
import Contador from "../../conceptos_basicos/Componentes-test/Contador";

describe("Prueba del componente contador", () => {
  let wrapper = shallow(<Contador />);

  beforeEach(() => {
    // se repite por que se necesita acceso a metodos
    wrapper = shallow(<Contador />);
  });

  test("Retorna el contenido html del componente Contador", () => {
    // const wrapper = shallow(<Contador />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Retorna el valor del número por defecto", () => {
    // const wrapper = shallow(<Contador />);

    const valuePorDefecto = wrapper.find("p").text();

    expect(valuePorDefecto).toBe("100");
  });

  test("Debe incrementar con el boton +1", () => {
    wrapper.find("button").at(0).simulate("click");

    const textWrapper = wrapper.find("p").text();

    expect(textWrapper).toBe("101");
  });

  test("Debe decrementar con el boton -1", () => {
    wrapper.find("button").at(1).simulate("click");

    const textWrapper = wrapper.find("p").text();
    expect(textWrapper).toBe("99");
  });

  test("debe reiniciar el contador al valor por defecto", () => {
    // const wrapper1 = shallow(<Contador valor={105} />);
    // console.log(wrapper);
    wrapper.find("button").at(1).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    wrapper.find("button").at(2).simulate("click");

    const counterText = wrapper.find("p").text();
    // console.log(reset);
    expect(counterText).toBe("100");
  });
});

// Notas:
// - comentado "wrapper" y usado de formal global en el "describe" para usarlo en todos los test
// - beforeEach reinicia el <Contador/> en cada ejecución de test
