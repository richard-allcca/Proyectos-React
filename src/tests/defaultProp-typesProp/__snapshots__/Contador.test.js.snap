// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente contador Retorna el contenido html del componente Contador 1`] = `
<Fragment>
  <h2>
    Contador
  </h2>
  <section
    className="contador"
  >
    <button
      onClick={[Function]}
    >
      +
    </button>
    <p>
      100
    </p>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </section>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
